extensions/azure/data-plane/data-factory/src/test/java/org/eclipse/dataspaceconnector/azure/dataplane/azuredatafactory/AzureDataFactoryTransferRequestValidatorTest.java

```diff
@@ -0,0 +1,144 @@
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataFlowRequest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.pipeline.AzureStorageTestFixtures.createAccountName;
+import static org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.pipeline.AzureStorageTestFixtures.createBlobName;
+import static org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.pipeline.AzureStorageTestFixtures.createContainerName;
+import static org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.pipeline.AzureStorageTestFixtures.createDataAddress;
+import static org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.pipeline.AzureStorageTestFixtures.createRequest;
+import static org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.pipeline.AzureStorageTestFixtures.createSharedKey;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+class AzureDataFactoryTransferRequestValidatorTest {
+    static final Faker FAKER = new Faker();
+
+    DataAddress.Builder source = createDataAddress(AzureBlobStoreSchema.TYPE);
+    DataAddress.Builder destination = createDataAddress(AzureBlobStoreSchema.TYPE);
+
+    static Map<String, String> sourceProperties = Map.of(
+            AzureBlobStoreSchema.ACCOUNT_NAME, createAccountName(),
+            AzureBlobStoreSchema.CONTAINER_NAME, createContainerName(),
+            AzureBlobStoreSchema.BLOB_NAME, createBlobName(),
+            AzureBlobStoreSchema.SHARED_KEY, createSharedKey());
+
+    static Map<String, String> destinationProperties = Map.of(
+            AzureBlobStoreSchema.ACCOUNT_NAME, createAccountName(),
+            AzureBlobStoreSchema.CONTAINER_NAME, createContainerName(),
+            AzureBlobStoreSchema.SHARED_KEY, createSharedKey());
+
+    static DataFlowRequest.Builder request = createRequest(AzureBlobStoreSchema.TYPE);
+
+    static DataFlowRequest requestWithProperties = request
+            .sourceDataAddress(createDataAddress(AzureBlobStoreSchema.TYPE).properties(sourceProperties).build())
+            .destinationDataAddress(createDataAddress(AzureBlobStoreSchema.TYPE).properties(destinationProperties).build())
+            .build();
+
+    String extraKey = FAKER.lorem().word();
+    String extraValue = FAKER.lorem().word();
+
+    AzureDataFactoryTransferRequestValidator validator = new AzureDataFactoryTransferRequestValidator();
+
+    @ParameterizedTest(name = "{index} {0}")
+    @MethodSource("canHandleArguments")
+    void canHandle(String ignoredName, String sourceType, String destinationType, boolean expected) {
```

Comment:
nit: tests naming in this suite seems to be inconsistent. 
 ```suggestion
    void validate_canHandle(String ignoredName, String sourceType, String destinationType, boolean expected) {
```

Reply:
The methods are called after the [method under test] optionally followed by a condition tested.

Added the condition even in parameterised tests to make this clearer.